parsing/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -3948,6 +3948,84 @@ SqlTypeNameSpec TypeName() :
     }
 }
 
+/**
+* Parse datetime types: date, time, timestamp.
+* Override the function in core parser.
+*/
+SqlTypeNameSpec DateTimeTypeName() :
+{
+    int precision = -1;
+    final SqlTypeName typeName;
+    SqlTimeZoneOption timeZoneOpt = SqlTimeZoneOption.WITHOUT_TIME_ZONE;
+}
+{
+    <DATE> {
+        typeName = SqlTypeName.DATE;
+        return new SqlBasicTypeNameSpec(typeName, getPos());
+    }
+|
+    LOOKAHEAD(2)
+    <TIME>
+    precision = PrecisionOpt()
+    timeZoneOpt = TimeZoneOption()
+    {
+        switch (timeZoneOpt) {
+           case WITHOUT_TIME_ZONE:
```

Comment:
nit: Can we just define the WITHOUT_TIME_ZONE case immediately above the default case with a fallthrough?

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/216/commits/4c4612dab93e39794b54dc478f3f52a64438567d